source/test/TestApp/GDK/StepTimer.h

```diff
@@ -0,0 +1,189 @@
+//
+// StepTimer.h - A simple timer that provides elapsed time information
+//
+
+#pragma once
+
+#include <cmath>
+#include <exception>
+#include <stdint.h>
+
+namespace DX
+{
+    // Helper class for animation and simulation timing.
+    class StepTimer
+    {
+    public:
+        StepTimer() noexcept(false) :
+            m_elapsedTicks(0),
+            m_totalTicks(0),
+            m_leftOverTicks(0),
+            m_frameCount(0),
+            m_framesPerSecond(0),
+            m_framesThisSecond(0),
+            m_qpcSecondCounter(0),
+            m_isFixedTimeStep(false),
+            m_targetElapsedTicks(TicksPerSecond / 60)
+        {
+            if (!QueryPerformanceFrequency(&m_qpcFrequency))
+            {
+                throw std::exception( "QueryPerformanceFrequency" );
+            }
+
+            if (!QueryPerformanceCounter(&m_qpcLastTime))
+            {
+                throw std::exception( "QueryPerformanceCounter" );
+            }
+
+            // Initialize max delta to 1/10 of a second.
+            m_qpcMaxDelta = static_cast<uint64_t>(m_qpcFrequency.QuadPart / 10);
+        }
+
+        // Get elapsed time since the previous Update call.
+        uint64_t GetElapsedTicks() const					{ return m_elapsedTicks; }

```

Comment:
>	 [](start = 41, length = 1)

nit: tabs.  might want to search in visual studio code,etc and remove them all

Reply:
Will do.